numpy/lib/shape_base.py

```diff
@@ -858,15 +858,15 @@ def tile(A, reps):
         return _nx.array(A, copy=True, subok=True, ndmin=d)
     else:
         c = _nx.array(A, copy=False, subok=True, ndmin=d)
-    shape = list(c.shape)
-    n = max(c.size, 1)
     if (d < c.ndim):
         tup = (1,)*(c.ndim-d) + tup
-    for i, nrep in enumerate(tup):
-        if nrep != 1:
-            c = c.reshape(-1, n).repeat(nrep, 0)
-        dim_in = shape[i]
-        dim_out = dim_in*nrep
-        shape[i] = dim_out
-        n //= max(dim_in, 1)
+    shape = tuple(s*t for s, t in zip(c.shape, tup))
+    n = c.size
+    if n != 0:
+        for i, nrep in enumerate(tup):
+            if nrep != 1:
+                c = c.reshape(-1, n).repeat(nrep, 0)
+            n //= shape[i]/nrep
```

Comment:
Look at the original code, where the division was always by the original, unmodified dimension. You are doing something different here, which likely explains the test failures.